hazelcast-jet-core/src/main/java/com/hazelcast/jet/Sources.java

```diff
@@ -92,38 +96,106 @@ private Sources() {
      * processors will leverage data locality by fetching only those entries
      * that are stored on the member where they are running.
      * <p>
+     * The number of Hazelcast partitions should be configured to at least
+     * {@code localParallelism * clusterSize}, otherwise some processors will
+     * have no partitions assigned to them.
+     * <p>
+     * This source does not save any state to snapshot. If the job is restarted,
+     * all entries will be emitted again.
+     * <p>
      * If the {@code IMap} is modified while being read, or if there is a
      * cluster topology change (triggering data migration), the source may
      * miss and/or duplicate some entries.
      */
-    public static <K, V> Source<Map.Entry<K, V>> readMap(String mapName) {
+    public static <K, V> Source<Map.Entry<K, V>> readMap(@Nonnull String mapName) {
         return fromProcessor("readMap(" + mapName + ')', SourceProcessors.readMapP(mapName));
     }
 
     /**
      * Returns a source that fetches entries from the Hazelcast {@code IMap}
      * with the specified name, filters them using the supplied predicate,
      * transforms them using the supplied projection function, and emits
-     * the results. Its processors will leverage data locality by fetching
-     * only those entries that are stored on the member where they are running.
+     * them as {@code Map.Entry}. Its processors will leverage data locality
+     * by fetching only those entries that are stored on the member where they
+     * are running.
+     * <p>
+     * The number of Hazelcast partitions should be configured to at least
```

Comment:
This is too low-level. A user of the Pipeline API isn't exposed to "local parallelism". If we decide we really have to mention it, it should be phrased differently.